### ISSUE TITLE
[Drag Drop] Mouse cursor disappears on left click when minDragTime/minDragDistance are set above 0

### ISSUE THREAD

**Issue by tarnos12** [ivan-mogilko/ags-script-modules#1]
**Describe the bug**
Setting `DragMinDistance` and `DragMinTime` above 0 in order to allow left clicking items without instantly dragging them, causes the mouse cursor to disappear(the graphic)

**To Reproduce**
In DragDrop script function `ResetToDefaults` set those properties to any number above 0, preferably big enough to give yourself time to left click before the drag/drop happens.

**Expected behavior**
The cursor should never disappear when left mouse button is released.

**Comment by ivan-mogilko** [comment]
@tarnos12 I was unable to reproduce this problem.

Used demo game [from here](https://github.com/ivan-mogilko/ags-script-modules/tree/master/demos/DragDropDemo). Set `DragMinDistance` and `DragMinTime` to some larger values in room4's `room_AfterFadeIn`. Clicking or failing to drag an inventory item does not cause cursor graphic to dissapear.

Upon double checking the code, the module does anything to mouse cursor or cursor's visible state at all. Therefore if it changes, that must be a script in your game.

**Comment by tarnos12** [comment]
@ivan-mogilko You are correct, but there is still a bug.
Here is my code:
```
function UpdateCursorGraphic()
{
	// Hide/Show cursor based on if we drag an item or not.
	if(DragDropCommon._InvItem == null)
	{
		Mouse.Visible = true;
	}
	else
	{
		// We are dragging an item, so hide the cursor.
		Mouse.Visible = false;
	}
}

function repeatedly_execute()
{
	UpdateCursorGraphic();
}
```

The problem is that even if I just left click(without dragging an item) the `DragDropCommon._InvItem` is set to an item and never cleared(because drag/drop didn't happen)

**Comment by ivan-mogilko** [comment]
The _InvItem and similar attributes are set at the moment of detecting an object under cursor, even if the dragging did not start yet, because it needs to know what is tried to be dragged.

The DragDrop module passes following events:
* EvtWantObject - where it looks for an object under cursor
* EvtObjectHooked - when it registered an object
* EvtDragStarted - when it actually began dragging.

The _InvItem is set between EvtWantObject and EvtObjectHooked.
The DragMinDistance and DragMinTime are tested between EvtObjectHooked and EvtDragStarted.

There are also following DragDrop flags that tell about current state:
* HasObjectHooked - if an object was found
* IsDragging - if an object is actually being dragged.

So the correct test for a dragged action should test current state and the object:
```if(DragDrop.IsDragging && DragDropCommon._InvItem != null)```
or only `IsDragging`, if you're only dragging inventory items in your game.

The fact that _InvItem is never reset is actually a mistake; at the moment this should be worked around by testing `IsDragging`.

**Comment by tarnos12** [comment]
Thanks, I will just use `IsDragging` for the time being.

**Comment by tarnos12** [comment]
@ivan-mogilko Btw, I was going to recommend creating a demo game for play testing, you might want to add the demo from above to the git page somewhere.

**Comment by ivan-mogilko** [comment]
>  you might want to add the demo from above to the git page somewhere.

The Demo Game is posted on the forum page: <https://www.adventuregamestudio.co.uk/forums/index.php?topic=53332.0>

**Comment by ivan-mogilko** [comment]
Okay, I see now, this module is missing a "cancel" event, that would help to find out if object was cancelled before dragging.

**Comment by tarnos12** [comment]
Actually ignore me, there already is "demo" folder, that's what I meant.

**Comment by ivan-mogilko** [comment]
Introduced EvtCancelled and fixed DragDropCommon module: 6634c722cffdf914badf448710e02bfe23d8e9ab
